app/src/main/java/woowacourse/shopping/di/Injector.kt

```diff
@@ -0,0 +1,5 @@
+package woowacourse.shopping.di
+
+interface Injector {
+    fun <T : Any> inject(modelClass: Class<T>): T
+}
```

Comment:
이름은 Injector이지만 정작 하는 일은 Module 객체와 협력해서 인자로 받은 클래스의 인스턴스를 반환해주네요 😢 
따라서 저라면
```suggestion
interface InstanceProvider{
    fun <T : Any> getInstance(modelClass: Class<T>): T
}
```
이런 식으로 특정 클래스의 인스턴스를 제공한다는 의미를 가지도록 만들 것 같아요!

Reply:
그런 시각도 말이 되는 것 같네요! 
하지만, 저는 좀 더 DI 라이브러리의 성격을 가진 '주입자'라는 의미의 Injector가 좀 더 선호되는 것 같습니다. 실제로 내부적으로 재귀를 돌면서, modelClass를 만들기 위한 인자들을 구성한 다음에 modelClass의 주생성자에 인자들을 주입해서 만들고 있기 때문입니다. 

추가적으로 이런 예시가 있더라고요!
[Angular inject 함수 공식 문서](https://angular.io/api/core/inject)

Reply:
그렇군요! 특정 클래스의 인스턴스를 생성하는 함수의 이름을 `inject`라고 지어도, DI 라이브러리라는 배경을 생각한다면 `inject`라는 함수가 무슨 목적으로 쓰이는지 알 수 있겠네요 😄